python/beeai_framework/adapters/litellm/chat.py

```diff
@@ -95,17 +105,22 @@ async def _create(
         return response_output
 
     async def _create_stream(self, input: ChatModelInput, _: RunContext) -> AsyncGenerator[ChatModelOutput]:
+        # TODO: handle tool calling for streaming
         litellm_input = self._transform_input(input)
         parameters = litellm_input.model_dump()
         parameters["stream"] = True
         response = await acompletion(**parameters)
 
-        # TODO: handle tool calling for streaming
+        is_empty = True
         async for chunk in response:
+            is_empty = False
             response_output = self._transform_output(chunk)
-            if not response_output:
-                continue
-            yield response_output
+            if response_output:
+                yield response_output
+
+        if is_empty:
+            # TODO: issue https://github.com/BerriAI/litellm/issues/8868
+            raise ChatModelError("Stream response is empty.")
```

Comment:
Note: this causes e2e `test_multi_agents_workflow_creation` in `test_multi_agents.py` to fail here. Also when run with `test_example_execution` from #420 the following examples also fail here: `basic.py`, `tools/decorator.py`, `tools/duckduckgo.py`, `tools/openmeteo.py`. It is also worth noting that all of these don't fail when run individually (even the test case), only when run as a pytest suite.

Reply:
`agents/bee.py`, `agents/granite.py`, and`agents/simple.py` also fail here, but the error shows differently since you added error handling to their `__main__`

Reply:
Also upon further testing the following failing's stack trace also leads back to this
```
FAILED tests/examples/test_examples.py::test_example_execution[workflows/multi_agents.py] - beeai_framework.agents.errors.AgentError: Maximal amount of global retries (0) has been reached.
```